process/wsclient/client.go

```diff
@@ -0,0 +1,166 @@
+package wsclient
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/data/typeConverters/uint64ByteSlice"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/data"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/gorilla/websocket"
+)
+
+type wsConn interface {
+	io.Closer
+	ReadMessage() (messageType int, p []byte, err error)
+	WriteMessage(messageType int, data []byte) error
+}
+
+var (
+	log           = logger.GetOrCreate("process/wsclient")
+	retryDuration = time.Second * 5
+)
+
+type client struct {
+	urlReceive string
+	actions    map[data.OperationType]func(marshalledData []byte) error
+}
+
+func NewWebSocketClient(urlReceive string, actions map[data.OperationType]func(marshalledData []byte) error) (*client, error) {
```

Comment:
missing comment

Reply:
it could have been named `New` since you'd have `wsclient.New()` instead of `wsclient.NewWebSocketClient()` which is redundant

Reply:
renamed

Reply:
and added comment